### ISSUE TITLE
FlyController maintain up

### ISSUE THREAD

**Issue by Korijn** [pygfx/pygfx#509]
The flycontroller rolls about its viewing axis while rotating the view with the mouse. That's not supposed to be the case with a fly camera.

**Comment by almarklein** [comment]
How do you mean? It should only roll around its view axis when q or e are pressed.

What does happen is then when you e.g. pitch up and then rotate left, and then pitch back, the horizon is no longer horizontal. You'd need to correct using q/e. Or we can add automatic-roll to keep `up` up.

**Comment by almarklein** [comment]
If we would keep the horizon level (as with the orbit controller) things would become unstable in near vertical flights, and loopings may not be possible. 

When flying a plane this happens too, and you'd have to adjust the roll. It's been too long since I played Descent .... but nowadays you can play all these games in the browser so I tried and this is what it does: the behavior looks like what our controller does, but when you're rotated it automatically rolls the ship. Except it rolls to the nearest 90 degree reference axis. I did something similar in the FlyCamera in visvis, except it always pulls you towards the ground.

**Comment by Korijn** [comment]
Maybe I was expecting first person controls, instead of actual airplane flight controls.

**Comment by almarklein** [comment]
Good point. Thinking about Minecraft creative mode. I guess both have their use ... a bit similar to orbit vs trackball, I suppose. We could have a property to toggle between these two modes. Or a subclass.

**Comment by Korijn** [comment]
Well, maybe we add a FirstPersonController?
